src/main/java/http/request/HttpRequest.java

```diff
@@ -73,4 +82,31 @@ public boolean loggedIn() {
     public String getParameter(String key) {
         return this.requestLine.getParameter(key);
     }
+
+    public HttpSession getSession() {
+        if (sessionIsNull()) {
+            this.session = new HttpSession(UUID.randomUUID().toString());
+        }
+        return this.session;
+    }
+
+    public boolean sessionIsNull() {
+        return Objects.isNull(session);
+    }
+
+    private static HttpSession findSession(RequestHeader requestHeader, Map<String, HttpSession> sessionMap) {
+        RequestCookie cookie = requestHeader.getRequestCookie();
```

Comment:
java 1.8 에서는 Optional 을 이용할 수 있어 간단한 예제 코드를 첨부해요
```suggestion
return Optional.ofNullable(requestHeader.getRequestCookie())
                .map(cookie -> cookie.get(SESSION_ID))
                .map(sessionMap::get)
                .orElse(null);
```